snuba/query/processors/uuid_column_processor.py

```diff
@@ -0,0 +1,159 @@
+from typing import List
+import uuid
+
+from snuba.query.conditions import (
+    binary_condition,
+    ConditionFunctions,
+    FUNCTION_TO_OPERATOR,
+    is_condition,
+)
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.logical import Query
+from snuba.query.matchers import (
+    AnyOptionalString,
+    Or,
+    Param,
+    String,
+)
+from snuba.query.matchers import Column as ColumnMatch
+from snuba.query.matchers import FunctionCall as FunctionCallMatch
+from snuba.query.matchers import Literal as LiteralMatch
+from snuba.query.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class UUIDColumnProcessor(QueryProcessor):
+    """
+    If a condition is being performed on a column that stores UUIDs (as defined in the constructor)
+    then change the condition to use a proper UUID instead of a string.
+    """
+
+    def formatted_uuid_pattern(self, suffix: str = "") -> FunctionCallMatch:
+        return FunctionCallMatch(
+            Param("format_alias" + suffix, AnyOptionalString()),
+            String("replaceAll"),
+            (
+                FunctionCallMatch(
+                    None,
+                    String("toString"),
+                    (
+                        Param(
+                            "formatted_uuid_column" + suffix,
+                            ColumnMatch(None, None, self.__uuid_column_match),
+                        ),
+                    ),
+                ),
+            ),
+            with_optionals=True,
+        )
+
+    def uuid_column_pattern(self, suffix: str = "") -> Param:
+        return Param(
+            "uuid_column" + suffix, ColumnMatch(None, None, self.__uuid_column_match)
+        )
+
+    def __init__(self, uuid_columns: List[str]) -> None:
+        self.__unique_uuid_columns = set(uuid_columns)
```

Comment:
why can't this be a set to begin with ? So people cannot pass duplicates even if they want.